ios/FluentUI/Navigation/SearchBar.swift

```diff
@@ -419,6 +445,27 @@ open class SearchBar: UIView {
         NSLayoutConstraint.activate(constraints)
     }
 
+    private func setupLayoutForLeadingView() {
+        guard let leadingView = leadingView else {
+            textFieldLeadingConstraint?.constant = Constants.searchTextFieldLeadingInset
+            return
+        }
+
+        let leadingViewSize = leadingView.sizeThatFits(searchTextFieldBackgroundView.frame.size)
+
+        textFieldLeadingConstraint?.constant = Constants.searchIconInset + leadingViewSize.width + Constants.searchTextFieldLeadingInset
```

Comment:
It seems like a better approach to tie the trailing constraint of the icon to the leading anchor of the badge view, and the trailing anchor of the badge view to the leading anchor of the text field.

Reply:
Done